CRM/Contribute/Form/Contribution/Main.php

```diff
@@ -1304,7 +1304,12 @@ public function assignFormVariablesByContributionID() {
     $paymentBalance = CRM_Contribute_BAO_Contribution::getContributionBalance($this->_ccid);
     //bounce if the contribution is not pending.
     if ((int) $paymentBalance <= 0) {
-      CRM_Core_Error::statusBounce(ts("Returning since contribution has already been handled."));
+      CRM_Core_Error::statusBounce(ts("This contribution is already fully paid."));
+    }
+    $contribution = civicrm_api3('Contribution', 'getsingle', array('id' => $this->_ccid));
+    if ($contribution['contribution_status'] == 'Pending' || $contribution['contribution_status'] == 'Partially paid') {
```

Comment:
I think you can't use 'contribution_status' since its a label field(correct me if i am wrong) which can be changed from UI.

Reply:
`If (in_array($contribution['contribution_status'], ['Pending', 'Partially Paid'])) {`